lib/node_modules/@stdlib/array/base/map/lib/assign.js

```diff
@@ -0,0 +1,164 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+
+
+// FUNCTIONS //
+
+/**
+* Applies a callback function to elements in an input array and assigns results to elements in an output array.
+*
+* @private
+* @param {Collection} x - input array
+* @param {Collection} y - output array
+* @param {integer} strideX - stride length for x
+* @param {NonNegativeInteger} offsetX - starting index for x
+* @param {Function} fcn - callback function
+* @param {*} thisArg - callback execution context
+* @returns {Collection} output array
+*
+* @example
+* var ones = require( '@stdlib/array/base/ones' );
+* var zeros = require( '@stdlib/array/base/zeros' );
+*
+* function scale( x ) {
+*     return x * 10.0;
+* }
+*
+* var x = ones( 4 );
+* var y = zeros( x.length );
+* var out = internal( x, y, 1, 0, scale );
+* // returns [ 10.0, 10.0, 10.0, 10.0 ]
+*
+* var bool = ( out === y );
+* // returns true
+*/
+function internal( x, y, strideX, offsetX, fcn, thisArg ) {
+	var io;
+	var i;
+
+	io = offsetX;
+	for ( i = 0; i < y.length; i++ ) {
```

Comment:
You should not be basing the loop on `y`! Given that we support a stride and offset for `y`, the `length` of `y` could be much greater than `x`. This needs to be fixed here and below.

This should have been caught via unit tests, which suggests there is a gap in testing.